addons/pkg/util/addon_util.go

```diff
@@ -218,3 +229,57 @@ func IsPackageInstallPresent(ctx context.Context,
 
 	return true, nil
 }
+
+// AddFinalizerToCRD adds finalizer to the config CRD if not present and
+// returns true if finalizer is added
+func AddFinalizerToCRD(log logr.Logger, addonName string, configCRD client.Object) bool {
+
+	var patchAddonSecret bool
+
+	// add finalizer to addon secret
+	if !controllerutil.ContainsFinalizer(configCRD, addontypes.AddonFinalizer) {
+		log.Info("Adding finalizer to addon secret", constants.AddonNameLogKey, addonName)
+		controllerutil.AddFinalizer(configCRD, addontypes.AddonFinalizer)
+		patchAddonSecret = true
+	}
+
+	return patchAddonSecret
+}
+
+// RemoveFinalizerFromCRD removes finalizer from the config CRD if present and
+// returns true if finalizer is removed
+func RemoveFinalizerFromCRD(log logr.Logger, addonName string, configCRD client.Object) bool {
```

Comment:
nit: I am assuming you meant `addonConfigCR` or `addonConfig` ?

Reply:
changed to ```addonCR``` for a more generic name. How do you think @danniel1205 ?